ens-improvement-proposals/ensip-14-platform-source-parameter.md

```diff
@@ -0,0 +1,57 @@
+---
+description: Using the reveal secret as a way to have on chain information about the source of the registration
+---
+
+# ENSIP-14: On Chain Source Parameter
+
+| **Author**    | Alex Slobo (@slobo.eth) and Alex Van de Sande (Avsa.eth)
+| ------------- | ------------------------------------------------------------------------------------------------------------------------------- |
+| **Status**    | Draft                                                                                                                           |
+| **Submitted** | 2023-02-07                                                                                                                      |
+
+## Abstract
+
+We have many reasons to measure user registration names. Knowing our ecosystem and which apps are popular is essential for identifying the creators who contribute. Retroactive funding via grants is often better than proactive funding, as it allows us to detect what has already worked, rather than relying on guesses. This community often sees unexpected successes that we can't predict.
+
+At the same time, it's important to make the process transparent, open source, and privacy-preserving. Source data would also help create referral programs.
+
+### Specification
+
+The ETH Registrar Controller takes a random secret as part of its input. This secret parameter is random data that obscures the name before the final step of the registration process is complete. It is 32 bytes of entropy and the ens manager app on app.ens.domains generates it by calling `require('crypto').randomBytes(32)`.
+
+This proposal suggests a social convention that generates the secret from a combination of the first **4 bytes of the namehash**, plus another **4 bytes for user referrals**, combined with the appropriate number of random bytes.
+
+The first 16 chars of the secret would be the platform name and the next 14 would be used as an identifier of the user referral. For example, if the above was registered via ensfairy, the secret would be:
```

Comment:
This is still very ambiguous. The secret is a binary value; 'chars' refers to characters in text. No encoding is specified. Instead, you should talk about how many bytes it is, independent of the encoding of it.

Reply:
Also, you mention 'the next 14', but it looks to me like you're only using 16 hex characters (8 bytes).

Reply:
Fixed the "next 14" error and clarified that by chars I meant in hex format.

I have an easier time thinking about hexes because that's the format that most people will see it, it's how it's on etherscan, on Metamask and it's how I see it when I am developing a frontend. 

I understand some more technical engineers like yourself would rather think about what the specification is and therefore prefer bits and bytes. I think it's important to keep both for clarity then.